frontend/src/pages/CreateProfile/BasicInfo.js

```diff
@@ -0,0 +1,178 @@
+import React, { Component } from 'react';
+import styles from '../../styles/CreateProfile.module.css';
+import Validator from 'react-forms-validator';
+
+class BasicInfo extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            firstName: '',
+            lastName: '',
+            emailAddress: '',
+            password: '',
+            phoneNumber: '',
+            confirmPassword: '',
+            website: '',
+        };
+        this.errors = true;
+    }
+
+    componentDidUpdate(_prevProps, _prevState) {
+        if (this.props.clickedNext) {
+            this.props.onSubmitData(this.state, this.errors);
+        }
+        return null;
+    }
+
+    componentDidMount() {
+        if (this.props.savedData !== null) {
+            this.setState(this.props.savedData);
+            this.errors = false;
+        }
+    }
+
+    handleFirstNameChange = event => {
```

Comment:
you don't have to do this, but to make it cleaner you can refactor all these functions that do the exact same thing into one:
```
handleChange = inputField => event => {
    this.setState({ [inputField]: event.target.value });
}
```

then for example, you could replace your onChange to `onChange={this.handleChange('lastName')}`